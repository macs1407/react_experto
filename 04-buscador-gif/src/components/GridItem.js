import React from 'react';
import PropTypes from 'prop-types';

const GridItem = ({title, url}) => {
    return ( 
        <div className="card animate__bounceOutUp">
            <img src={url} title={title} alt={title}/>
            <p>{title}</p>
        </div>
     );
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
 
export default GridItem;