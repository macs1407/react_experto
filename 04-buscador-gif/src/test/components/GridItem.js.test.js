import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GridItem from '../../components/GridItem';

describe('Pruebas de la aplicacion <GridItem />', () => {

    const title = "Un titulo";
    const url = "htttp://www.imagen.com/logo.png";

    const wraper = shallow(<GridItem 
        title={title}
        url={url}
      />);
 
    test('Prueba para veriricar que el componente suba bien ', () => {       
    
        expect(wraper).toMatchSnapshot();
    });

    test('Debe tener un titulo ', () => {       
        const titulo = wraper.find('p');
        expect(titulo.text().trim()).toBe(title);
    });


    test('Debe tener la imagen igual al url ', () => {       
        const imagen = wraper.find('img');
        expect(imagen.prop('src')).toBe(url);
        expect(imagen.prop('alt')).toBe(title);
    });
    
})
