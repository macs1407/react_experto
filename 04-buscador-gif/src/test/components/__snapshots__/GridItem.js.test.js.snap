// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de la aplicacion <GridItem /> Prueba para veriricar que el componente suba bien  1`] = `
<div
  className="card animate__bounceOutUp"
>
  <img
    alt="Un titulo"
    src="htttp://www.imagen.com/logo.png"
    title="Un titulo"
  />
  <p>
    Un titulo
  </p>
</div>
`;
